internal/http/check.go

```diff
@@ -0,0 +1,210 @@
+			Report: report.Report{
+				CheckData: report.CheckData{
+					CheckID:          job.CheckID,
+					StartTime:        job.StartTime, // TODO: Is this correct or should be time.Now()
+					ChecktypeName:    c.config.Check.CheckTypeName,
+					ChecktypeVersion: c.config.Check.CheckTypeVersion,
+					Options:          job.Options,
+					Target:           job.Target,
+				},
+				ResultData: report.ResultData{},
+			},
+		},
+	}
+
+	runtimeState := state.State{
+		ResultData:       &checkState.state.Report.ResultData,
+		ProgressReporter: state.ProgressReporterHandler(checkState.SetProgress),
+	}
+	logger.WithField("opts", job.Options).Info("Starting check")
+	err = c.checker.Run(ctx, job.Target, job.AssetType, job.Options, runtimeState)
+	c.checker.CleanUp(ctx, job.Target, job.AssetType, job.Options)
+	checkState.state.Report.CheckData.EndTime = time.Now()
+	elapsedTime := time.Since(startTime)
+	// If an error has been returned, we set the correct status.
+	if err != nil {
+		if errors.Is(err, context.Canceled) {
+			checkState.state.Status = agent.StatusAborted
+		} else if errors.Is(err, state.ErrAssetUnreachable) {
+			checkState.state.Status = agent.StatusInconclusive
+		} else if errors.Is(err, state.ErrNonPublicAsset) {
+			checkState.state.Status = agent.StatusInconclusive
+		} else {
+			logger.WithError(err).Error("Error running check")
+			checkState.state.Status = agent.StatusFailed
+			checkState.state.Report.Error = err.Error()
+		}
+	} else {
+		checkState.state.Status = agent.StatusFinished
+	}
+	checkState.state.Report.Status = checkState.state.Status
+
+	logger.WithFields(log.Fields{"seconds": elapsedTime.Round(time.Millisecond * 100).Seconds(), "state": checkState.state.Status}).Info("Check finished")
+
+	// Initialize sync point for the checker and the push state to be finished.
+	out, err := json.Marshal(checkState.state)
+	if err != nil {
+		logger.WithError(err).Error("error marshalling the check state")
+		http.Error(w, "error marshalling the check state", http.StatusInternalServerError)
+		return
+	}
+	w.Write(out)
+}
+
+// RunAndServe implements the behavior needed by the sdk for a check runner to
+// execute a check.
+func (c *Check) RunAndServe() {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		if c.shuttedDown == nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.Write([]byte(`"OK"`))
+		} else {
+			// the server is shutting down
+			http.Error(w, "shuttingDown", http.StatusServiceUnavailable)
+		}
+	})
+	http.HandleFunc("/run", c.ServeHTTP)
+	c.Logger.Info(fmt.Sprintf("Listening at %s", c.server.Addr))
+	go func() {
+		if err := c.server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			c.Logger.WithError(err).Error("Starting http server")
+		}
+		c.Logger.Info("Stopped serving new connections.")
+		if c.shuttedDown != nil {
+			c.Logger.Info("Notifying shuttedDown")
+			c.shuttedDown <- 1
+			c.Logger.Info("Notified shuttedDown")
+		}
```

Comment:
With the proposed changes I think this can be replaced with:

```suggestion
		c.done <- true
```

Reply:
done 877c330718594bef1468b09846b8dbdfe4e3f388